### PR TITLE
Plan behavioural test suite migration to rstest-bdd

### PR DESCRIPTION
**Pull request by leynos**
## Summary
- Plan the migration of behavioural tests to rstest-bdd; introduce ADR-002: Replace cucumber-rs with rstest-bdd for behavioural tests.
- Adds a clear migration plan and aligns roadmap tasks to guide the upcoming port of behavioural suites.
- This PR focuses on planning and documentation; concrete code migrations will follow in subsequent PRs.

## Changes
- Added docs/adr-002-replace-cucumber-with-rstest-bdd.md outlining the rationale, decision drivers, considered options, and a phased implementation plan.
- Updated docs/roadmap.md with a concrete sub-task outline for replacing cucumber-rs behavioural tests with rstest-bdd (Option 3).
- Minor Makefile adjustment: set PUBLISH_CHECK_FLAGS to empty (previously --allow-dirty).

## Rationale
- Align behavioural tests with the rest of the workspace by using rstest-bdd, enabling fixture sharing and ScenarioState while removing the bespoke async runner and World model.
- Reduce dependencies and maintenance burden by dropping cucumber, gherkin, and the Tokio-based test runtime from dev-dependencies.
- Improve ergonomics and typing in step implementations via rstest-bdd features (typed placeholders, data tables, and tag filtering).

## Migration plan (high level)
Phase 0 – Discovery and parity baseline
- Catalogue existing behavioural coverage and dependencies
- Freeze baseline by running make test and archive cucumber outputs

Phase 1 – Introduce rstest-bdd scaffolding
- Add rstest-bdd dev-dependencies to ortho_config and hello_world crates
- Create shared test_support::bdd scaffolding exposing fixtures and helpers
- Add a canary scenario to validate integration

Phase 2 – Replace worlds with fixtures and ScenarioState
- Port World structs to rstest fixtures and ScenarioState/Slot<T>
- Introduce synchronous wrappers around async helpers where needed

Phase 3 – Port ortho_config step modules and features
- Translate cucumber step modules to rstest-bdd macros
- Bind scenarios to tests via scenarios! / #[scenario]
- Remove cucumber.rs once migrated
- Update behavioural docs

Phase 4 – Port the hello_world example suite
- Mirror Phase 3 approach for the example
- Bind all scenarios with appropriate feature flags and tag filtering

Phase 5 – Clean-up and docs alignment
- Remove cucumber/gherkin and Tokio runtime from dev-deps
- Update docs and CHANGELOG to reflect the migration
- Update CI/docs onboarding with rstest-bdd guidance

## Acceptance criteria
- All .feature files are executed via rstest-bdd; no crate declares a [[test]] with harness = false for behavioural coverage.
- Behavioural fixtures rely on rstest (and ScenarioState) rather than bespoke World structs.
- Optional scenarios gated by tags compile correctly via macro tags expressions.
- Documentation and examples reference rstest-bdd and reflect the new workflow.

## Testing plan
- Run cargo test for ortho_config and examples/hello_world (including all feature-gated paths)
- Run make test, make lint, and make check-fmt to catch regressions introduced by the migration plan.

## Notes for reviewers
- This PR is primarily planning and documentation-oriented to establish ADR-002 and the migration roadmap. Concrete code migrations will be implemented in follow-up PRs.
- Please review the proposed strategy and edge cases (tag filtering, fixtures sharing, and breaking changes in step definitions).

## Related issues
- ADR-002: Replace cucumber with rstest-bdd for behavioural tests (documented in docs/adr-002-replace-cucumber-with-rstest-bdd.md)
- Roadmap alignment: see docs/roadmap.md for migration tasks and milestones.

🌿 Generated by [Terry](https://www.terragonlabs.com)

---

ℹ️ Tag @terragon-labs to ask questions and address PR feedback

📎 **Task**: https://www.terragonlabs.com/task/4af8dd83-2e03-4b00-aeb7-a4eb0d6009e5

## Summary by Sourcery

Plan and document the replacement of the cucumber-rs behavioural test suite with rstest-bdd, including an ADR, updated roadmap tasks, and a minor Makefile configuration tweak.

Enhancements:
- Plan and document the migration of behavioural tests from cucumber-rs to rstest-bdd by introducing ADR-002 and outlining a phased implementation plan.
- Align the project roadmap with the migration strategy by adding detailed sub-tasks for porting tests, fixtures, and feature files to rstest-bdd.

Build:
- Clear PUBLISH_CHECK_FLAGS in the Makefile to remove the default --allow-dirty option.

Documentation:
- Add ADR-002 document detailing decision drivers, considered options, and a multi-phase migration plan.
- Update docs/roadmap.md with tasks for replacing cucumber-rs behavioural tests with rstest-bdd.

### CODE REVIEW

File: docs/roadmap.md [hunk 1]
```diff
@@ -277,6 +277,25 @@ references the relevant design guidance.
...
+
+  - [ ] Port every module under `ortho_config/tests/steps` to
+    `rstest_bdd_macros`, bind the existing feature files via `scenarios!` or
+    `#[scenario]`, and delete the bespoke `tests/cucumber.rs` runner.
+
+  - [ ] Migrate the `hello_world` example suite: convert the world helpers to
+    `rstest` fixtures, rebind the feature file using compile-time tag filters,
+    and drop the `tests/cucumber` harness plus its `[[test]]` entry.
+
+  - [ ] Remove the `cucumber`/`gherkin` dev-dependencies, clean the unused
```
[the remark this answers is not in the document]
Comment (leynos) [reply docs/roadmap.md]: Has this now been resolved?
